client/src/components/Header/ShareSessionButton.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { FC } from 'react';
+import {
+  Avatar,
+  Box,
+  Button,
+  IconButton,
+  ListItemIcon,
+  ListItemText,
+  Menu,
+  MenuItem,
+} from '@mui/material';
+import { Link, Share, Visibility } from '@mui/icons-material';
+import { useSnackbar } from 'notistack';
+import { useAppStore } from '~/store/app';
+import lightTheme from '~/styles/theme';
+import { useLocation } from 'react-router-dom';
+
+const ShareSessionButton: FC<unknown> = () => {
+  const { enqueueSnackbar } = useSnackbar();
+  const location = useLocation();
+  const windowIsSmall = useAppStore((state) => state.windowIsSmall);
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const [copySuccess, setCopySuccess] = React.useState({});
+  const openShareMenu = Boolean(anchorEl);
+
+  // Adds and removes '/view' from the URL when necessary
+  const composeUrl = (viewOnly?: boolean): string => {
+    let viewOnlyUrlEnding = location.hash;
+    // Remove trailing /
+    while (viewOnlyUrlEnding.endsWith('/')) {
+      viewOnlyUrlEnding = viewOnlyUrlEnding.slice(0, -1);
+    }
+    if (viewOnly && !viewOnlyUrlEnding.includes('/view')) {
+      viewOnlyUrlEnding += '/view';
+    }
+    if (!viewOnly && viewOnlyUrlEnding.includes('/view')) {
+      const viewStringIndex = viewOnlyUrlEnding.indexOf('/view');
+      viewOnlyUrlEnding = viewOnlyUrlEnding.substring(0, viewStringIndex);
+    }
+    return `${window.location.origin}${window.location.pathname}${viewOnlyUrlEnding}`;
```

Comment:
```suggestion
        return `${window.location.origin}${(window.location.pathname + '/').replace(/\/\/$/, '')}${viewOnlyUrlEnding}`;
```

You might be able to just do something like this.  A trailing slash here should be superficial to inferno, but meaningful to the browser... so this will 'toggle' the trailing slash.

It might be a bit dangerous, as slashes are meaningful to relative links (things like `../something`), but i don't think we have any relative links in this interface?  And if you think this might be something we can do more properly in a planned react router update then this would let us get this in.

I did this real quick, it probably could be cleaner.

Reply:
Could also do the same with `?` but this was a little easier.

Reply:
This does work -- there should be no relative links after the session ID so this is safe to use. Thanks, this was helpful!